### PR TITLE
Add ability to always stack resizable stones

### PR DESCRIPTION
**Pull request by raiyni**
Closes https://github.com/runelite/runelite/issues/1189

Unlike https://github.com/runelite/runelite/pull/5780, there is no visible relayout when the window is resized because the layout is done in the script.

### CODE REVIEW

File: runelite-client/src/main/scripts/LayoutResizableStones.rs2asm [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+   iconst                 10551325
+   enum
+   if_getwidth
+   sub
+   iconst                 429
+   if_icmplt              LABEL29
+
+   iconst                 0                  ; should resizable stones be forced to stack
+   sconst                 "forceStackStones" ; push event name
+   runelite_callback     					 ; invoke callback
```
Comment (Nightfirecat) [inline runelite-client/src/main/scripts/LayoutResizableStones.rs2asm]: You've got some tabs here that should be spaces instead

File: runelite-client/src/main/scripts/LayoutResizableStones.rs2asm [hunk 2]
```diff
@@ -0,0 +1,225 @@
+.id                 920
+.int_stack_count    2
+.string_stack_count 0
+.int_var_count      5
+.string_var_count   0
```
Comment (abextm) [inline runelite-client/src/main/scripts/LayoutResizableStones.rs2asm]:
```suggestion
.string_var_count   0
; callback "forceStackStones"
;   Used by the InterfaceStylesPlugin to enable it's Always stack bottom bar option
;   Toggle the option when you have the bottom line top level interface on and your screen is large enough for the stones to be in a single line
```

**Comment by abextm** [comment]
This script needs to be rebased

**Comment by raiyni** [comment]
rebased script

File: runelite-client/src/main/java/net/runelite/client/plugins/interfacestyles/InterfaceStylesConfig.java [hunk 3]
```diff
@@ -72,4 +72,14 @@ default boolean rsCrossSprites()
 	{
 		return false;
 	}
+
+	@ConfigItem(
+		keyName = "alwaysStack",
+		name = "Always stack bottom bar",
+		description = "Always stack the bottom bar in resizable"
+	)
+	default boolean stack()
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/interfacestyles/InterfaceStylesConfig.java]:
```suggestion
	default boolean alwaysStack()
```

File: runelite-client/src/main/java/net/runelite/client/plugins/interfacestyles/InterfaceStylesPlugin.java [hunk 4]
```diff
@@ -106,6 +107,19 @@ public void onConfigChanged(ConfigChanged config)
...
 	}
 
+	@Subscribe
+	public void onScriptCallbackEvent(ScriptCallbackEvent event)
+	{
+		String eventName = event.getEventName();
+		int[] intStack = client.getIntStack();
+		int intStackSize = client.getIntStackSize();
+
+		if (config.stack() && "forceStackStones".equals(eventName))
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/interfacestyles/InterfaceStylesPlugin.java]:
```suggestion
		if ("forceStackStones".equals(eventName) && config.alwaysStack())
```

File: runelite-client/src/main/java/net/runelite/client/plugins/interfacestyles/InterfaceStylesPlugin.java [hunk 5]
```diff
@@ -106,6 +107,19 @@ public void onConfigChanged(ConfigChanged config)
 		}
 	}
 
+	@Subscribe
+	public void onScriptCallbackEvent(ScriptCallbackEvent event)
+	{
+		String eventName = event.getEventName();
+		int[] intStack = client.getIntStack();
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/interfacestyles/InterfaceStylesPlugin.java]: these can be scoped in the if.

**Pull request merged**